packages/core/src/utils/index.ts

```diff
@@ -0,0 +1,7 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/core
```

Comment:
I am proposing to move these utility functions and types to a new package `@loopback/utils`, so that we can start using type definitions like `Constructor` in places like `@loopback/rest` too.